### PR TITLE
pr-9 add Order model:

### PR DESCRIPTION
**Pull request by nikolasdymchenko7**
- added scripts for adding tables:
  - orders
  - order_items
- added models: Order and OrderItem
- added repositories, mappers, dto(s) and service for the models
- added 4 new endpoints in controller OrderController:
  - createOrder
  - updateOrderStatus
  - getOrdersByUserId
  - getOrderItemByOrderId
  - getOrderItem

### CODE REVIEW

File: src/main/java/com/store/book/model/Order.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "orders")
+public class Order {
```
Comment (Rommelua) [inline src/main/java/com/store/book/model/Order.java:27]: image

File: src/main/java/com/store/book/model/OrderItem.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import java.util.Objects;
+import lombok.Data;
+
+@Entity
+@Data
+@Table(name = "order_items")
+public class OrderItem {
```
Comment (Rommelua) [inline src/main/java/com/store/book/model/OrderItem.java:18]: image

File: src/main/java/com/store/book/repository/cartitem/CartItemRepository.java [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 import com.store.book.model.CartItem;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 
 public interface CartItemRepository extends JpaRepository<CartItem, Long> {
+    @Modifying
+    @Query("DELETE FROM CartItem c WHERE c.shoppingCart.id = :shoppingCartId")
+    void deleteByShoppingCartIAndId(Long shoppingCartId);
```
Comment (Rommelua) [inline src/main/java/com/store/book/repository/cartitem/CartItemRepository.java:11]: image

File: src/main/java/com/store/book/service/order/OrderService.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.store.book.dto.orderitem.OrderItemResponseDto;
+import com.store.book.dto.orders.CreateOrderResponseDto;
+import com.store.book.dto.orders.OrderResponseDto;
+import com.store.book.dto.orders.UpdateOrderDto;
+import com.store.book.model.User;
+import java.util.List;
+
+public interface OrderService {
+    CreateOrderResponseDto createOrder(User user);
```
Comment (Rommelua) [inline src/main/java/com/store/book/service/order/OrderService.java:11]:
```suggestion
    OrderResponseDto createOrder(User user);
```

File: src/main/java/com/store/book/service/order/OrderService.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    UpdateOrderDto updateOrderStatus(Long orderId, UpdateOrderDto orderDto);
```
Comment (Rommelua) [inline src/main/java/com/store/book/service/order/OrderService.java:13]:
```suggestion
    OrderResponseDto updateOrderStatus(Long orderId, UpdateOrderDto orderDto);
```

**Review by kerrrusha — APPROVED** [review]
looks good

**Pull request merged**